sections/featured-product.liquid

```diff
@@ -0,0 +1,597 @@
+{% comment %}theme-check-disable TemplateLength{% endcomment %}
+{{ 'section-main-product.css' | asset_url | stylesheet_tag }}
+{{ 'section-featured-product.css' | asset_url | stylesheet_tag }}
+{{ 'component-accordion.css' | asset_url | stylesheet_tag }}
+{{ 'component-badge.css' | asset_url | stylesheet_tag }}
+{{ 'component-price.css' | asset_url | stylesheet_tag }}
+{{ 'component-rte.css' | asset_url | stylesheet_tag }}
+{{ 'component-slider.css' | asset_url | stylesheet_tag }}
+
+<link rel="stylesheet" href="{{ 'component-cart-notification.css' | asset_url }}" media="print" onload="this.media='all'">
+<link rel="stylesheet" href="{{ 'component-deferred-media.css' | asset_url }}" media="print" onload="this.media='all'">
+
+{%- assign product = all_products[section.settings.product] -%}
```

Comment:
**Possible bug:** Using `section.settings.product` directly will result in an error with forms.

<img src="https://user-images.githubusercontent.com/37168033/126928930-13472266-4239-45cf-87d3-178497ee25fa.png" width="380">

If you try to output `{{ section.settings.product.title }}`, you'll see that the ProductDrop is not empty.

Reply:
That's weird, but I guess for now we can use the assign and bring the bug to the platform team and see if they can fix it for a refactor later.